oddt/shape.py

```diff
@@ -0,0 +1,258 @@
+            Molecule to compute USR shape descriptor
+
+        Returns
+        -------
+        shape_descriptor : numpy array, shape = (12)
+            Array describing shape of molecule
+    """
+    return common_usr(molecule)[0]
+
+
+def usr_cat(molecule):
+    """Computes USRCAT shape descriptor based on
+        Adrian M Schreyer, Tom Blundell (2012). USRCAT: real-time ultrafast
+        shape recognition with pharmacophoric constraints. Journal of
+        Cheminformatics, 2012 4:27.
+        http://dx.doi.org/10.1186/1758-2946-4-27
+
+        Parameters
+        ----------
+        molecule : oddt.toolkit.Molecule
+            Molecule to compute USRCAT shape descriptor
+
+        Returns
+        -------
+        shape_descriptor : numpy array, shape = (60)
+            Array describing shape of molecule
+    """
+    all_atoms_shape, points = common_usr(molecule)
+    ctd, cst, fct, ftf = points
+    hydrophobic_shape = common_usr(
+        molecule, ctd, cst, fct, ftf, 'ishydrophobe')[0]
+    aromatic_shape = common_usr(molecule, ctd, cst, fct, ftf, 'isaromatic')[0]
+    acceptor_shape = common_usr(molecule, ctd, cst, fct, ftf, 'isacceptor')[0]
+    donor_shape = common_usr(molecule, ctd, cst, fct, ftf, 'isdonor')[0]
+
+    cat_shape = np.hstack((all_atoms_shape, hydrophobic_shape,
+                           aromatic_shape, acceptor_shape, donor_shape))
+
+    return np.nan_to_num(cat_shape)
+
+
+def electroshape(mol):
+    """Computes shape descriptor based on
+    Armstrong, M. S. et al. ElectroShape: fast molecular similarity
+    calculations incorporating shape, chirality and electrostatics.
+    J Comput Aided Mol Des 24, 789-801 (2010).
+    http://dx.doi.org/doi:10.1007/s10822-010-9374-0
+
+    Aside from spatial coordinates, atoms' charges are also used
+    as the fourth dimension to describe shape of the molecule.
+
+        Parameters
+        ----------
+            mol : oddt.toolkit.Molecule
+
+        Returns
+        -------
+            shape_descriptor : numpy array, shape = (15)
+                               Array describing shape of molecule
+    """
+    if (mol.atom_dict['coords'] == 0).all():
+        raise Exception('Molecule needs 3D coordinates')
+
+    mi = 25  # scaling factor converting electron charges to Angstroms
+
+    four_dimensions = np.column_stack((mol.atom_dict['coords'],
+                                       mol.atom_dict['charge'] * mi))
+
+    c1 = four_dimensions.mean(0)  # geometric centre of the molecule
+    distances_c1 = norm(four_dimensions - c1, axis=1)
+
+    c2 = four_dimensions[distances_c1.argmax()]  # atom position furthest from c1
+    distances_c2 = norm(four_dimensions - c2, axis=1)
+
+    c3 = four_dimensions[distances_c2.argmax()]  # atom position furthest from c2
+    distances_c3 = norm(four_dimensions - c3, axis=1)
+
+    vector_a = c2 - c1
+    vector_b = c3 - c1
+    vector_as = vector_a[:3]    # spatial parts of these vectors -
```

Comment:
Join the comment and put it before `vector_as` definition with additional blank lines around this group of vectors (as and bs).